client/src/components/pages/admin/DataVisualizationPage.js

```diff
@@ -0,0 +1,373 @@
+import React from 'react';
+import { connect } from 'react-redux';
+import { Link } from 'react-router';
+import Chart1 from './LineChart';
+import ProgressBar from 'react-bootstrap/lib/Progressbar';
+import Chart2 from './DonutChart';
+
+import { getUsers, getAlerts} from 'actions';
+
+require('./tableStyles.scss');
+
+
+class DataVisualizationPage extends React.Component {
+
+    constructor(props) {
+        super(props);
+        this.defaultState = {
+            width: 0,
+            
+        }
+
+        this.state = this.defaultState;
+    }
+
+    componentWillMount() {
+        this.props.getUsers();
+        this.props.getAlerts();
+    }
+
+    filterAdvocates(users) {
+        let advocates = users.filter(function(user) { return user.role == "advocate"});
+        return advocates;
+    }
+
+    filterAdmin(users){
+        let admin = users.filter(function(user) { return user.role == "admin"});
+        return admin;
+    }
+
+    filterProviders(users){
```

Comment:
These 3 functions can be refactored into a single function that takes the role string as the 2nd parameter.